src/main/configurations/Translate/Common/xsl/BuildError.xsl

```diff
@@ -2,15 +2,23 @@
     <xsl:output method="xml" version="1.0" encoding="UTF-8" indent="yes" omit-xml-declaration="yes"/>
     <xsl:param name="code" select="'TechnicalError'" as="xs:string" />
     <xsl:param name="reason" select="'Undefined error'" as="xs:string" />
+    <xsl:param name="cause" as="node()?" />
     <xsl:param name="details" select="''" as="xs:string" />
     <xsl:param name="detailsXml" as="node()?" />
     
     <xsl:template match="/">
         <error>
-            <code><xsl:value-of select="$code" /></code>
-            <reason><xsl:value-of select="$reason" /></reason>
-            <details><xsl:value-of select="$details" /></details>
-            <detailsXml><xsl:copy-of select="$detailsXml"/></detailsXml>
+            <xsl:choose>
+                <xsl:when test="//error">
+                    <cause><xsl:copy-of select="$cause" /></cause>
+                </xsl:when>
+                <xsl:otherwise>
```

Comment:
There is no condition here. The original element can stay as they where. The new cause element is simply an optional node() that can be added like detailsXml. It can be a simple when /error copy-of $cause.